src/weights.jl

```diff
@@ -682,6 +682,35 @@ function mean(A::AbstractArray, w::UnitWeights; dims::Union{Colon,Int}=:)
     return mean(A, dims=dims)
 end
 
+"""
+    mean(f, A::AbstractArray, w::AbstractWeights)
+
+Compute the weighted mean of array `A`, after transforming it'S
+contents with the function `f`, with weight vector `w` (of type
+`AbstractWeights`).
+
+# Examples
+```julia
+n = 20
+x = rand(n)
+w = rand(n)
+mean(√, x, weights(w))
+```
+"""
+mean(f, A::AbstractArray, w::AbstractWeights) =
+_funcweightedmean(f, A, w)
+
+function _funcweightedmean(f, A::AbstractArray, w::AbstractWeights)
+    return sum(Broadcast.broadcasted(f, A, w) do f, a_i, wg
+        return f(a_i) * wg
+    end) / sum(w)
```

Comment:
You should use `Broadcast.instantiate`. Moreover, to me it seems this should be changed to
```suggestion
    return sum(Broadcast.instantiate(Broadcast.broadcasted(A, w) do a_i, wg
        return f(a_i) * wg
    end)) / sum(w)
```

Reply:
Using `Broadcast.instantiate` causes extra allocations, no?

Reply:
I haven't tried. But without `Broadcast.instantiate` it will be slow and fall back to Cartesian indexing. See, e.g., https://github.com/JuliaLang/julia/pull/31020 ("we require Broadcast.instantiate for fast reduce").